### PR TITLE
0.4.2

### PR DESCRIPTION
**Pull request by AndreiKingsley**

### CODE REVIEW

File: kandy-api/src/main/kotlin/org/jetbrains/kotlinx/kandy/dsl/internal/DatasetHandler.kt [hunk 1]
```diff
@@ -62,11 +62,9 @@ public class DatasetHandler(
             referredColumns[name] = columnId
             name
         }
-        // TODO
```
Comment (devcrocod) [inline kandy-api/src/main/kotlin/org/jetbrains/kotlinx/kandy/dsl/internal/DatasetHandler.kt:65]: What's todo?

File: kandy-api/src/main/kotlin/org/jetbrains/kotlinx/kandy/dsl/internal/validation.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package org.jetbrains.kotlinx.kandy.dsl.internal
+
+import org.jetbrains.kotlinx.kandy.ir.aes.AesName
+
+public fun<T: Comparable<T>> checkInRange(aesName: AesName, value: T, range: ClosedRange<T>){
+    if (value !in range) {
```
Comment (devcrocod) [inline kandy-api/src/main/kotlin/org/jetbrains/kotlinx/kandy/dsl/internal/validation.kt:6]: requieres

**Review by devcrocod — CHANGES_REQUESTED** [review]

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/geomWrap.kt [hunk 3]
```diff
@@ -5,48 +5,53 @@
...
-        RECT -> org.jetbrains.letsPlot.Geom.rect()
-        RIBBON -> org.jetbrains.letsPlot.Geom.ribbon()
-        SEGMENT -> org.jetbrains.letsPlot.Geom.segment()
-        SMOOTH -> GeomOptions(GeomKind.SMOOTH)
-        STEP -> org.jetbrains.letsPlot.Geom.step()
-        TILE -> org.jetbrains.letsPlot.Geom.tile()
-        TEXT -> org.jetbrains.letsPlot.Geom.text()
-        //  VIOLIN -> GeomOptions(GeomKind.VIOLIN)
-        V_LINE -> org.jetbrains.letsPlot.Geom.vline()
+        is LetsPlotGeom -> when(this) {
```
Comment (devcrocod) [inline kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/geomWrap.kt:16]: What's LetsPlotGeom?
Why is function not called LetsPlotGeom but Geom?

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/layerWrap.kt [hunk 4]
```diff
@@ -26,14 +26,12 @@ internal fun Layer.wrap(
     }
     val addGroups = datasets[datasetIndex] is GroupedData
     val groupKeys = (datasets[datasetIndex] as? GroupedData)?.keys
-    //todo
-    /*val mappings = if (datasetIndex == 0) {
+    val mappings = if (inheritsBindings) {
         globalMappings + mappings
```
Comment (devcrocod) [inline kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/layerWrap.kt:30]: In what case is `else` needed?

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/scaleWrap.kt [hunk 5]
```diff
@@ -461,6 +466,25 @@ internal fun Scale.wrap(
                                 format = format,
                                 naValue = naValue
 
+                            )*/
+
+                            org.jetbrains.letsPlot.intern.Scale(
```
Comment (devcrocod) [inline kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/scaleWrap.kt:471]: qualified name

**Review by devcrocod — CHANGES_REQUESTED** [review]
For a pie chart always need to specify constant values for x and y?

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/layerWrap.kt (hunk 4 above)
Comment (AndreiKingsley) [reply kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/layerWrap.kt:30]: This flag is used if we do not inherit mappings. It is necessary if we change dataset - for example if we use stats, we don't have old columns, so the mappings are not valid.

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/geomWrap.kt (hunk 3 above)
Comment (AndreiKingsley) [reply kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/translator/geomWrap.kt:16]: There is a `Geom` interface in IR. `LetsPlotGeom` is implementation of it in lets-plot

**Pull request merged**